### PR TITLE
Enhance fakepg: send and recv

### PR DESCRIPTION
**Pull request by xmfan**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #107626
* __->__ #107625

from working on a starter task with @wanchaol (T161350434):
Enhance Fake Process Group by adding missing collectives: send, recv

### CODE REVIEW

**Review by fduwjj — APPROVED** [review]
LGTM

**Review by wanchaol — APPROVED** [review]
lgtm

File: test/distributed/test_fake_pg.py [hunk 1]
```diff
@@ -153,6 +153,21 @@ def test_alltoall_base(self):
         dist.all_to_all_single(out_tensor, in_tensor, output_split, input_split)
         self.assertEqual(tuple(out_tensor.shape), (3, 3))
 
+    def test_send(self):
+        store = FakeStore()
+        dist.init_process_group(backend="fake", rank=0, world_size=2, store=store)
+
+        tensor = torch.ones(3, 3)
+        dist.send(tensor, 1)
```
Comment (wanchaol) [inline test/distributed/test_fake_pg.py:161]: can you add some assertions on the shape of the send call too?

**Comment by xmfan** [comment]
removing diff reference from PR body

@pytorchbot merge

**Comment by xmfan** [comment]
unlinked internal diff
@pytorchbot merge

**Pull request closed without merging**